hydra-node/src/Hydra/Chain/Blockfrost/Client.hs

```diff
@@ -321,29 +322,85 @@ toCardanoGenesisParameters bfGenesis =
+mkEraHistory :: BlockfrostClientT IO EraHistory
+mkEraHistory = do
+  eras' <- Blockfrost.getNetworkEras
+  let eras = filter withoutEmptyEra eras'
+  let summary = mkEra <$> eras
+  case nonEmptyFromList summary of
+    Nothing ->
+      liftIO $ throwIO $ BlockfrostError "Failed to create EraHistory."
+    Just s -> pure $ EraHistory (mkInterpreter $ Summary s)
  where
-  Blockfrost.Genesis
-    { _genesisNetworkMagic
-    , _genesisSystemStart
-    , _genesisSlotLength
-    , _genesisEpochLength
-    } = genesis
-
-  summary :: Summary (CardanoEras StandardCrypto)
-  summary =
-    Summary . NonEmptyOne $
-      EraSummary
-        { eraStart = initBound
-        , eraEnd = EraUnbounded
-        , eraParams
-        }
-
-  eraParams =
+  mkBound Blockfrost.NetworkEraBound{_boundEpoch, _boundSlot, _boundTime} =
+    Bound
+      { boundTime = RelativeTime _boundTime
+      , boundSlot = SlotNo $ fromIntegral _boundSlot
+      , boundEpoch = EpochNo $ fromIntegral _boundEpoch
+      }
+  mkEraParams Blockfrost.NetworkEraParameters{_parametersEpochLength, _parametersSlotLength, _parametersSafeZone} =
     EraParams
-      { eraEpochSize = EpochSize $ fromIntegral _genesisEpochLength
-      , eraSlotLength = mkSlotLength $ fromIntegral _genesisSlotLength
-      , eraSafeZone = UnsafeIndefiniteSafeZone
-      , eraGenesisWin = GenesisWindow 1
+      { eraEpochSize = EpochSize $ fromIntegral _parametersEpochLength
+      , eraSlotLength = mkSlotLength _parametersSlotLength
+      , eraSafeZone = StandardSafeZone _parametersSafeZone
+      , eraGenesisWin = GenesisWindow _parametersSafeZone
+      }
+  mkEra Blockfrost.NetworkEraSummary{_networkEraStart, _networkEraEnd, _networkEraParameters} =
+    EraSummary
+      { eraStart = mkBound _networkEraStart
+      , eraEnd = EraEnd $ mkBound _networkEraEnd
+      , eraParams = mkEraParams _networkEraParameters
       }
+  withoutEmptyEra
+    Blockfrost.NetworkEraSummary
+      { _networkEraStart = Blockfrost.NetworkEraBound{_boundTime = boundStart}
+      , _networkEraEnd = Blockfrost.NetworkEraBound{_boundTime = boundEnd}
+      } = boundStart == 0 && boundEnd == 0
+
+----------------
+-- Wallet API --
+----------------
+
+-- | Query the Blockfrost API for address UTxO and convert to cardano 'UTxO'.
+queryUTxO :: SigningKey PaymentKey -> NetworkId -> BlockfrostClientT IO UTxO
+queryUTxO sk networkId = do
+  let address = Blockfrost.Address vkAddress
+  utxo <- Blockfrost.getAddressUtxos address
+  let cardanoAddress = mkVkAddress networkId vk
+  pure $ toCardanoUTxO utxo cardanoAddress
+ where
+  vk = getVerificationKey sk
+  vkAddress = textAddrOf networkId vk
+
+-- | Query the Blockfrost API for 'Genesis'
+queryGenesis :: BlockfrostClientT IO Blockfrost.Genesis
+queryGenesis = Blockfrost.getLedgerGenesis
+
+-- | Query the Blockfrost API for 'Genesis' and convert to cardano 'ChainPoint'.
+queryTip :: QueryPoint -> BlockfrostClientT IO ChainPoint
```

Comment:
Must: need a point to query a point

This function does not make sense (from its explanation and signature). If this is really a `queryTip`, it should have signature

```suggestion
queryTip :: BlockfrostClientT IO ChainPoint
```